RandLAPACK/drivers/rl_cqrrp.hh

```diff
@@ -269,40 +272,36 @@ int CQRRP_blocked<T, RNG>::call(
     T* Work2    = ( T * ) calloc( n, sizeof( T ) );
     //*******************POINTERS TO DATA REQUIRING ADDITIONAL STORAGE END*******************
 
-    T norm_A     = lapack::lange(Norm::Fro, m, n, A, lda);
-    T norm_A_sq  = std::pow(norm_A, 2);
-    T norm_R     = 0.0;
-    T norm_R11   = 0.0;
-    T norm_R12   = 0.0;
-    T norm_R_i   = 0.0;
-    T approx_err = 0.0;
-
     if(this -> timing) {
         preallocation_t_stop  = high_resolution_clock::now();
         preallocation_t_dur   = duration_cast<microseconds>(preallocation_t_stop - preallocation_t_start).count();
         saso_t_start = high_resolution_clock::now();
     }
 
-    // Skethcing in an embedding regime
-    RandBLAS::SparseDist DS = {.n_rows = d, .n_cols = m, .vec_nnz = this->nnz};
-    RandBLAS::SparseSkOp<T, RNG> S(DS, state);
-    state = RandBLAS::fill_sparse(S);
-
-    RandBLAS::sketch_general(
-        Layout::ColMajor, Op::NoTrans, Op::NoTrans,
-        d, n, m, (T) 1.0, S, 0, 0, A, lda, (T) 0.0, A_sk, d
-    );
+    if (this -> use_gaussian) {
+        T* S  = ( T * ) calloc( d * m, sizeof( T ) );
+        RandBLAS::DenseDist D(d, m);
+        state = RandBLAS::fill_dense(D, S, state).second;
+        blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, d, n, m, 1.0, S, d, A, m, 0.0, A_sk, d);
+        free(S);
+    } else {
+        // Skethcing in an embedding regime
+        RandBLAS::SparseDist DS = {.n_rows = d, .n_cols = m, .vec_nnz = this->nnz};
+        RandBLAS::SparseSkOp<T, RNG> S(DS, state);
+        state = RandBLAS::fill_sparse(S);
+
+        RandBLAS::sketch_general(
+            Layout::ColMajor, Op::NoTrans, Op::NoTrans,
+            d, n, m, (T) 1.0, S, 0, 0, A, lda, (T) 0.0, A_sk, d
+        );
+    }
```

Comment:
IMO, the implementation should only support dense sketching. You can have a flag for Gaussian if you want, but that should only distinguish between Gaussian and Uniform.

Reply:
Yeah, i know. 
Since i was working on the two PRs at the same time, many changes did not get reflected here.